scripts/validate_string_concatenation.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+"""
+GH #30454
+
+Check where there is a string that needs to be concatenated.
+
+This is necessary after black formating,
+where for example black transforms this:
+
+>>> foo = (
+...         "bar "
+...         "baz"
+...     )
+
+into this:
+
+>>> foo = ("bar " "baz")
+
+Black is not considering this as an
+issue (see issue https://github.com/psf/black/issues/1051),
+so we are checking it here.
+"""
+
+import argparse
+import os
+import sys
+import token
+import tokenize
+from typing import Dict, Generator, List, Tuple
+
+FILE_EXTENSIONS_TO_CHECK: Tuple[str, str, str, str] = (
+    ".py",
+    ".pyx",
+    ".pyx.ini",
+    ".pxd",
+)
+
+
+def main(*, source_path: str = ".", output_format: str) -> bool:
+    """
+    Main entry point of the script.
+
+    Parameters
+    ----------
+    source_path : str, default '.'
+        Source path representing path to a file/directory.
+    output_format : str
+        Output format of the script.
+
+    Returns
+    -------
+    bool
+        True if found any strings that needs to be concatenated.
+
+    Raises
+    ------
+    ValueError
+        If the `source_path` is not pointing to existing file/directory.
+    """
+    if not os.path.exists(source_path):
+        raise ValueError(
+            "Please enter a valid path, pointing to a valid file/directory."
+        )
+
+    is_failed: bool = False
+
+    if os.path.isfile(source_path):
+        for values in strings_to_concatenate(source_path):
+            is_failed = True
+            print(output_format.format(**values))
+
+    for subdir, _, files in os.walk(source_path):
+        for file_name in files:
+            if any(
+                file_name.endswith(extension) for extension in FILE_EXTENSIONS_TO_CHECK
+            ):
+                for values in strings_to_concatenate(os.path.join(subdir, file_name)):
+                    is_failed = True
+                    print(output_format.format(**values))
+    return is_failed
+
+
+def strings_to_concatenate(source_path: str) -> Generator[Dict[str, str], None, None]:
+    """
+    Yielding the strings that needs to be concatenated in a given file.
+
+    Parameters
+    ----------
+    source_path : str
+        File path pointing to a single file.
+
+    Yields
+    ------
+    Dict[str, str]
```

Comment:
```suggestion
    dict of {str: str}
```

In the docstrings we don't use mypy syntax